File: documentation/manual/detailedTopics/production/code/java/CustomSSLContextProvider.java
package java;

// #javaexample
import play.server.ApplicationProvider;
import play.server.SSLContextProvider;

import javax.net.ssl.SSLContext;
import java.security.NoSuchAlgorithmException;

public class CustomSSLContextProvider implements SSLContextProvider {

    @Override
    public SSLContext createSSLContext(ApplicationProvider applicationProvider) {
        try {
            // change it to your custom implementation
            return SSLContext.getDefault();
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }
}
// #javaexample

File: documentation/manual/detailedTopics/production/code/scala/CustomSSLContextProvider.scala
package scala

// #scalaexample
import javax.net.ssl.SSLContext
import play.core.ApplicationProvider
import play.server.api.SSLContextProvider

class CustomSSLContextProvider extends SSLContextProvider {

  override def createSSLContext(appProvider: ApplicationProvider): SSLContext = {
    // change it to your custom implementation
    SSLContext.getDefault
  }

}
// #scalaexample
